src/main/scala/mesosphere/marathon/core/storage/store/impl/memory/InMemoryPersistenceStore.scala

```diff
@@ -148,6 +149,24 @@ class InMemoryPersistenceStore(implicit
 
     Future.successful(Done)
   }
+
+  private[this] val migrationInProgress: AtomicBoolean = new AtomicBoolean(false)
+
+  override def startMigration(): Future[Done] = {
+    require(isOpen, "the store must be opened before it can be used")
+    if (!migrationInProgress.compareAndSet(false, true)) {
+      Future.failed(new IllegalStateException("Migration is already in progress"))
+    }
+    Future.successful(Done)
```

Comment:
I think we need to wrap this Future in `else` block otherwise we will always return success :)